ormar/queryset/queryset.py

```diff
@@ -972,26 +972,36 @@ async def get(self, *args: Any, **kwargs: Any) -> "T":  # noqa: CCR001
         self.check_single_result_rows_count(processed_rows)
         return processed_rows[0]  # type: ignore
 
-    async def get_or_create(self, *args: Any, **kwargs: Any) -> "T":
+    async def get_or_create(
+        self,
+        _defaults: Optional[Dict[str, Any]] = None,
+        *args: Any,
+        **kwargs: Any,
+    ) -> Tuple["T", bool]:
         """
         Combination of create and get methods.
 
         Tries to get a row meeting the criteria for kwargs
         and if `NoMatch` exception is raised
-        it creates a new one with given kwargs.
+        it creates a new one with given kwargs and _defaults.
 
         Passing a criteria is actually calling filter(*args, **kwargs) method described
         below.
 
         :param kwargs: fields names and proper value types
         :type kwargs: Any
-        :return: returned or created Model
-        :rtype: Model
+        :param _defaults: default values for creating object
+        :type _defaults: Optional[Dict[str, Any]]
+        :return: model instance and a boolean
+        :rtype: Tuple("T", bool)
         """
         try:
-            return await self.get(*args, **kwargs)
+            return await self.get(*args, **kwargs), False
         except NoMatch:
-            return await self.create(**kwargs)
+            if _defaults is None:
+                return await self.create(**kwargs), True
+            else:
+                return await self.create(**kwargs, **_defaults), True
```

Comment:
please use:
```python
_defaults = _defaults or {}
```
to avoid the if statement.

The order of arguments is wrong -> `_defaults` should be first, but still overwritten by `kwargs` value if explicitly provided.

Also, this will break if `kwargs` and `_default` have the same parameter. 
You are missing a test that checks this (that would fail). 
Even if you provide defaults you can as a user still pass a different value for this parameter, the way you did it would always overwrite it with the default one.

so it should be:
```python
return await self.create(**{**_defaults, **kwargs}), True
```
(update _defaults with kwargs and only then unpack).

Reply:
should i raise an error if `_defaults` and `kwargs` have something in common?

Reply:
I think this is fine for `_defaults` and `kwargs` to have some keys in common.  
It means get object with the given `kwargs`, if object was not found, then create it with given `_defaults` and `kwargs` but `_defaults` should overwrite `kwargs`.  
What do you think about this implementation?

Reply:
OK, I forgot tit's get or create.
We can overwrite kwargs with defaults (and it will be in line with what django is doing).

Yes, it is fine to have the same key in kwargs and defaults (in your implementation currently they cannot), and we can agree defaults should overwrite the provided kwargs.

Also, I asked you to remove the if statement (the fewer conditionals the better), so you will always have defaults populated.

So full implementation will be like this:
```python
except NoMatch:
    _defaults = _defaults or {} # instantiate empty dict if defaults not provided)
    return await self.create(**{**kwargs, **_defaults}), True  # update kwargs dict with defaults and then unpack as kwargs to create
```

Reply:
Yeah I was waiting for your response.  
I'll make the changes and let you know.

Reply:
Changes have been made. @collerek